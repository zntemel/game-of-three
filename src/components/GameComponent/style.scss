.game-component {
  display: flex;
  flex-direction: column;
  width: 100%;
  overflow: auto;
  .ant-list-items {
    display: flex;
    flex-direction: column;
  }
  .game-number {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    font-size: 16px;
    margin: 10px;
    border-radius: 12px;
    padding: 10px 20px;
    width: 25%;
  }
  .computer,
  .playerOne {
    color: #000;
    background: #efefef;
    justify-content: flex-start !important;
  }
  .player,
  .playerTwo {
    color: #fff;
    background: #32a2f2;
    align-self: flex-end;
  }
}
