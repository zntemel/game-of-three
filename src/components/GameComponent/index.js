import React from "react";
import PropTypes from "prop-types";
import { List } from "antd";
import "./style.scss";

function GameComponent(props) {
  return (
    <div className="game-component">
      <List
        dataSource={props.turnArray}
        renderItem={(turn) => (
          <List.Item key={turn.id} className={"game-number " + turn.player}>
            <div>{turn.player}:</div>
            <div>{turn.value}</div>
          </List.Item>
        )}
      />
    </div>
  );
}

GameComponent.propTypes = {
  turnArray: PropTypes.array,
};

export default GameComponent;
