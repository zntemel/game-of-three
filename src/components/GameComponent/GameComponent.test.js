import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import GameComponent from "./index";

const turnArray = [
  {
    id: 0,
    player: "playerOne",
    value: 20,
  },
  {
    id: 1,
    player: "playerTwo",
    value: 7,
  },
  {
    id: 2,
    player: "playerOne",
    value: 2,
  },
  {
    id: 3,
    player: "playerTwo",
    value: 1,
  },
];

describe("Game Component", () => {
  const wrapper = shallow(<GameComponent turnArray={turnArray} />);
  const instance = wrapper.instance();

  describe("shallow wrapper", () => {
    it("should render succesfully with snapshots", () => {
      const component = renderer.create(wrapper).toJSON();
      expect(component).toMatchSnapshot();
    });

    it("instance should be null because of it's dumy component", () => {
      expect(instance).toEqual(null);
    });

    it("list should render with correctly", () => {
      expect(wrapper.find("List").props().dataSource).toEqual(turnArray);
    });
  });
});
