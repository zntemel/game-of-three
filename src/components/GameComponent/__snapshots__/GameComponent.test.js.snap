// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Component shallow wrapper should render succesfully with snapshots 1`] = `
<div
  className="game-component"
>
  <div
    className="ant-list ant-list-split"
  >
    <div
      className="ant-spin-nested-loading"
    >
      <div
        className="ant-spin-container"
      >
        <ul
          className="ant-list-items"
        >
          <li
            className="ant-list-item game-number playerOne"
          >
            <div>
              Computer
              :
            </div>
            <div>
              20
            </div>
          </li>
          <li
            className="ant-list-item game-number playerTwo"
          >
            <div>
              Computer
              :
            </div>
            <div>
              7
            </div>
          </li>
          <li
            className="ant-list-item game-number playerOne"
          >
            <div>
              Computer
              :
            </div>
            <div>
              2
            </div>
          </li>
          <li
            className="ant-list-item game-number playerTwo"
          >
            <div>
              Computer
              :
            </div>
            <div>
              1
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
